Brokerages/SymbolPropertiesDatabaseSymbolMapper.cs

```diff
@@ -0,0 +1,177 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Provides the mapping between Lean symbols and brokerage symbols using the symbol properties database
+    /// </summary>
+    public class SymbolPropertiesDatabaseSymbolMapper : ISymbolMapper
+    {
+        private readonly string _market;
+
+        // map Lean symbols to symbol properties
+        private readonly Dictionary<Symbol, SymbolProperties> _symbolPropertiesMap;
+
+        // map brokerage symbols to Lean symbols
+        private readonly Dictionary<string, Symbol> _symbolMap;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="SymbolPropertiesDatabaseSymbolMapper"/> class.
+        /// </summary>
+        /// <param name="market">The Lean market</param>
+        public SymbolPropertiesDatabaseSymbolMapper(string market)
+        {
+            _market = market;
+
+            var symbolPropertiesList =
+                SymbolPropertiesDatabase
+                    .FromDataFolder()
+                    .GetSymbolPropertiesList(_market)
+                    .Where(x => !string.IsNullOrWhiteSpace(x.Value.MarketTicker))
+                    .ToList();
+
+            _symbolPropertiesMap =
+                symbolPropertiesList
+                    .ToDictionary(
+                        x => Symbol.Create(x.Key.Symbol, x.Key.SecurityType, x.Key.Market),
+                        x => x.Value);
+
+            _symbolMap =
+                _symbolPropertiesMap
+                    .ToDictionary(
+                        x => x.Value.MarketTicker,
+                        x => x.Key);
+        }
+
+        /// <summary>
+        /// Converts a Lean symbol instance to a brokerage symbol
+        /// </summary>
+        /// <param name="symbol">A Lean symbol instance</param>
+        /// <returns>The brokerage symbol</returns>
+        public string GetBrokerageSymbol(Symbol symbol)
```

Comment:
I find confusing that this method is named Get...Symbol but does return a ticker name and not a Symbol.